modules/sitemaps/sitemaps.php

```diff
@@ -1,782 +1,1556 @@
+ *    of a tree of sitemap and sitemap index xml files, depending on the number of URLs.
  *
- * @param string $mysql_date UTC datetime in MySQL syntax of YYYY-MM-DD HH:MM:SS.
+ *    By default the sitemap contains published posts of type 'post' and 'page', as
+ *    well as the home url. To include other post types use the 'jetpack_sitemap_post_types'
+ *    filter.
  *
- * @return string ISO 8601 UTC datetime string formatted as YYYY-MM-DDThh:mm:ssTZD where timezone offset is always +00:00.
- */
-function jetpack_w3cdate_from_mysql( $mysql_date ) {
-	return str_replace( ' ', 'T', $mysql_date ) . '+00:00';
-}
-
-/**
- * Get the maximum comment_date_gmt value for approved comments for the given post_id.
+ * @link http://sitemaps.org/protocol.php Base sitemaps protocol.
+ * @link https://support.google.com/webmasters/answer/178636 Image sitemap extension.
  *
- * @module sitemaps
+ * 2. news-sitemap.xml
+ *    The news sitemap is generated on the fly when requested. It does not aim for
+ *    completeness, instead including at most 1000 of the most recent published posts
+ *    from the previous 2 days, per the news-sitemap spec.
  *
- * @param int $post_id Post identifier.
+ * @link http://www.google.com/support/webmasters/bin/answer.py?answer=74288 News sitemap extension.
  *
- * @return string datetime MySQL value or null if no comment found.
+ * @author Automattic
  */
-function jetpack_get_approved_comments_max_datetime( $post_id ) {
-	global $wpdb;
 
-	return $wpdb->get_var( $wpdb->prepare( "SELECT MAX(comment_date_gmt) FROM $wpdb->comments WHERE comment_post_ID = %d AND comment_approved = '1' AND comment_type=''", $post_id ) );
-}
+require dirname( __FILE__ ) . '/sitemap-buffer.php';
+require dirname( __FILE__ ) . '/sitemap-stylist.php';
+require dirname( __FILE__ ) . '/sitemap-logger.php';
 
 /**
- * Return the content type used to serve a Sitemap XML file.
- * Uses text/xml by default, possibly overridden by jetpack_sitemap_content_type filter.
- *
- * @module sitemaps
+ * Governs the generation, storage, and serving of sitemaps.
  *
- * @return string Internet media type for the sitemap XML.
+ * @since 4.5.0
  */
-function jetpack_sitemap_content_type() {
+class Jetpack_Sitemap_Manager {
+
 	/**
-	 * Filter the content type used to serve the XML sitemap file.
+	 * Maximum size (in bytes) of a sitemap xml file.
 	 *
-	 * @module sitemaps
+	 * @link http://www.sitemaps.org/
+	 * @since 4.5.0
+	 */
+	const SITEMAP_MAX_BYTES = 10485760; // 10485760 (10MB)
+
+	/**
+	 * Maximum size (in url nodes) of a sitemap xml file.
 	 *
-	 * @since 3.9.0
+	 * @link http://www.sitemaps.org/
+	 * @since 4.5.0
+	 */
+	const SITEMAP_MAX_ITEMS = 50000; // 50k
+
+	/**
+	 * Maximum size (in url nodes) of a news sitemap xml file.
 	 *
-	 * @param string $content_type By default, it's 'text/xml'.
+	 * @link https://support.google.com/news/publisher/answer/74288?hl=en
+	 * @since 4.5.0
 	 */
-	return apply_filters( 'jetpack_sitemap_content_type', 'text/xml' );
```

Comment:
Did you deprecate this filter? If so, you can add it to the list [here](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L5510).